api/authentication.py

```diff
@@ -45,10 +45,15 @@ def authenticate_user(request: Request) -> Optional[ResponseReturnValue]:
             # Redirect to the OIDC login page if not logged in
             if not oidc.user_loggedin:
                 # Copied from oidc.require_login decorator
-                redirect_uri = "{login}?next={here}".format(
-                    login=url_for("oidc_auth.login"),
-                    here=quote_plus(request.url),
-                )
+                next_url = request.args.get('next', '')
+                next_url = next_url.replace('\\', '')
+                if not urlparse(next_url).netloc and not urlparse(next_url).scheme:
```

Comment:
`urlparse` was renamed to `urllib.parse` in Python 3, it should be updated to that and also imported

https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse